pkg/controller/common.go

```diff
@@ -1130,3 +1129,36 @@ func WatchObject(mgr manager.Manager, objKind source.Kind) error {
 	})
 	return err
 }
+
+// WatchCRD creates a noop controller to set up a watch and an informer for CRD objects in controller-runtime manager
+func CacheCRD(mgr manager.Manager) error {
+	c, err := controller.New("CRD-controller", mgr, controller.Options{
+		Reconciler: &noOpReconciler{},
+	})
+	if err != nil {
+		return err
+	}
+
+	c.GetLogger().Info("Starting to watch CRDs")
+
+	// Watch for CRUD events on CRD objects
+	err = c.Watch(&source.Kind{Type: &apiext.CustomResourceDefinition{}}, &handler.EnqueueRequestForObject{}, predicate.Funcs{
+		CreateFunc: func(_ event.CreateEvent) bool {
+			return false
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			crdObj, ok := e.ObjectNew.(*apiext.CustomResourceDefinition)
+			if ok && (crdObj.Name == consts.AzVolumeAttachmentCRDName || crdObj.Name == consts.AzVolumeCRDName) && crdObj.DeletionTimestamp != nil {
+				return true
```

Comment:
We might want to dispatch a goroutine to update the atomic value representing `isDriverUninstall` like we do for `isRecoveryComplete` instead of listing all CRDs to determine if the driver is being uninstalled or not.

https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/95813ab637a8b79a0646701d94209929fa3a159e/pkg/controller/shared_state.go#L118